src/TextArea.tsx

```diff
@@ -0,0 +1,65 @@
+import clsx from 'clsx'
+import {
+	forwardRef,
+	InputHTMLAttributes,
+	useContext,
+	useRef,
+	useState,
+} from 'react'
+import { ThemeContext } from './context/ThemeContext'
+import { mergeRefs } from './utils/warning'
+import Label from './Label'
+
+type TextareaProps = InputHTMLAttributes<HTMLTextAreaElement> & {
+	invalid?: boolean
+	className?: string
+	disabled?: boolean
+	label?: string
+	name?: string
+}
+
+const TextArea = forwardRef<HTMLTextAreaElement, TextareaProps>(
+	(props, ref) => {
+		const { className, invalid, disabled, label, name, ...other } = props
+		const { textarea } = useContext(ThemeContext)
+		const textareaRef = useRef<HTMLTextAreaElement>(null)
+		const [value, setValue] = useState(props.value || '')
+
+		const baseStyles = textarea.base
+		const invalidStyles = textarea.invalid
+		const disabledStyles = textarea.disabled
+
+		const hasValidation = invalid !== undefined
+		const validationStyle = hasValidation
+			? invalid
+				? invalidStyles
+				: null
+			: null
+
+		const cls = clsx(
+			baseStyles,
+			// don't apply disabledStyle if has valid
+			!hasValidation && disabled && disabledStyles,
+			validationStyle,
+			className,
+		)
+		const textareaName = name ? name : 'textarea'
+
+		return (
+			<div className="relative">
+				{label && <Label htmlFor={textareaName}>{label}</Label>}
```

Comment:
`htmlFor` attribute should have a corresponding `id` on the `textarea`. Right now, if you click on the `label`, it does not focus on the `textarea`.

To fix this issue: you can add an `id` with the same name.

But I would prefer if you can remove the name `label` attribute from this component. And follow the approach we had for `input`.  

https://github.com/kirandash/greenhouse-react-ui/blob/ee3f1fbf606dff272e40a9fa76bedaa532123e4f/src/stories/Label.stories.jsx#LL29C1-L36C25

Please update the story, test case to support this.

Also you can remove the `name` attribute. User can choose to add a name attribute of their own. I am trying to maintain a consistent behaviour with the `input` component.